packages/components/pie-checkbox-group/src/index.ts

```diff
@@ -1,20 +1,108 @@
-import { LitElement, html, unsafeCSS } from 'lit';
-import { RtlMixin, defineCustomElement } from '@justeattakeaway/pie-webc-core';
-
+import {
+    LitElement, html, unsafeCSS, nothing, PropertyValues,
+} from 'lit';
+import { property } from 'lit/decorators.js';
+import {
+    RtlMixin, defineCustomElement, FormControlMixin, validPropertyValues,
+} from '@justeattakeaway/pie-webc-core';
+import { ifDefined } from 'lit/directives/if-defined.js';
 import styles from './checkbox-group.scss?inline';
-import { CheckboxGroupProps } from './defs';
+import { CheckboxGroupProps, defaultProps, statusTypes } from './defs';
 
 // Valid values available to consumers
 export * from './defs';
 
 const componentSelector = 'pie-checkbox-group';
+const assistiveTextId = 'assistive-text';
 
 /**
  * @tagname pie-checkbox-group
  */
-export class PieCheckboxGroup extends RtlMixin(LitElement) implements CheckboxGroupProps {
+export class PieCheckboxGroup extends FormControlMixin(RtlMixin(LitElement)) implements CheckboxGroupProps {
```

Comment:
Another generic question 😄 

Are we planning to handle checkbox nesting levels as part of checkbox group? So for consumers that want folder structure groupings, where the indeterminant status could be used?

Reply:
Not on this stage I think, since we don't have any designs for that type of the group. But we can work on it later if there will be the use case for that

Reply:
I think it'd be worth bringing up with design, as I can see this being a common use-case in back-office tools (and will likely mean we just need to set the structure of the slot markup for it). Obviously a separate ticket though, just wanted to check  😄  

cc @RLumleyJET

Reply:
We discussed this with Rachel and for now we won't add it until we get a request for this